File: 257. Binary Tree Paths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void trav(TreeNode node, ArrayList<String> arr, String curr){
        if(node == null){
            return;
        }

        if(node.left == null && node.right == null){
            arr.add(curr + node.val);
            return;
        }

        curr += String.format("%d", node.val) + "->";
        if(node.left != null) trav(node.left, arr, curr);
        if(node.right != null) trav(node.right, arr, curr);
    }
    public List<String> binaryTreePaths(TreeNode root) {
        ArrayList<String> arr = new ArrayList<>();
        trav(root, arr, "");
        return arr;
    }
}

File: 257. Binary Tree Paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void trav(TreeNode* node, vector<string>& arr, string curr) {
        if (node == nullptr) return;

        if (node->left == nullptr && node->right == nullptr) {
            arr.push_back(curr + to_string(node->val));
            return;
        }

        curr += to_string(node->val) + "->";
        if (node->left) trav(node->left, arr, curr);
        if (node->right) trav(node->right, arr, curr);
    }

    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> arr;
        trav(root, arr, "");
        return arr;
    }
};
